packages/client/src/schema/query.ts

```diff
@@ -313,12 +314,17 @@ export class Query<Record extends XataRecord, Result extends XataRecord = Record
    */
   getMany<Options extends RequiredBy<QueryOptions<Record>, 'columns'>>(
     options: Options
-  ): Promise<SelectedPick<Record, typeof options['columns']>[]>;
+  ): Promise<RecordArray<SelectedPick<Record, typeof options['columns']>>>;
+
+  async getMany<Result extends XataRecord>(options: QueryOptions<Record> = {}): Promise<RecordArray<Result>> {
+    const page = await this.getPaginated(options);
+
+    if (page.hasNextPage() && options.pagination?.size === undefined) {
+      console.debug('Calling getMany does not return all results. Paginate to get all results or call getAll.');
```

Comment:
debug is less noisy, but does not provide the easiest way to jump from console as trace. If docs are well explained and we recommend getPaginated, this could be enough.